src/debug/ee/debugger.cpp

```diff
@@ -16732,6 +16732,18 @@ void *DebuggerHeap::Alloc(DWORD size)
     ret = pCanary->GetUserAddr();
 #endif
 
+#ifdef FEATURE_PAL
+    // We don't have executable heap in PAL, but we still need to allocate 
+    // executable memory, that's why have change protection level for 
+    // each allocation. 
+    // TODO: We need to look how JIT solves this problem.
+    DWORD unusedFlags;
+    if (!VirtualProtect(ret, size, PAGE_EXECUTE_READWRITE, &unusedFlags))
```

Comment:
We would also want to remove the executable permissions when freeing that memory. Otherwise we would leak the permission to blocks allocated from the general heap after freeing block allocated from the debugger heap. 
And it seems we will need to add the executable protection in the `DebuggerHeap::Realloc` too.